sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -106,12 +103,16 @@ case class InsertIntoHiveTable(
         }
 
         writerContainer
-          .getLocalFileWriter(row, table.schema)
-          .write(serializer.serialize(outputData, standardOI))
+          .getLocalFileWriter(row, schema)
+          .write(newSer(fileSinkConf.getTableInfo).serialize(outputData, standardOI))
       }
 
       writerContainer.close()
-    }
+    }: Unit
+
+    writerContainer.driverSideSetup()
+    sc.sparkContext.runJob(rdd, sc.sparkContext.clean(writeToFile))
+    writerContainer.commitJob()
```

Comment:
I made these changes to clean the closure and avoid referencing the `$out` so that we don't need to add a lot of `@ transient`. However, it fails a test `CliSuite.Commands using SerDe provided in --jars` with timeout. Does anybody know the reason?